### DISCUSSION TITLE
Select value as object instead of string

### DISCUSSION THREAD

**Discussion by Arsikod** [react-hook-form/react-hook-form#6209] [Q&A] [answered]
Hey guys, is there any way to put select option value as an object (for example {name: 'John', age: 35}) and obtain it in data object upon submit? I try to pass object but get "[Object Object]" as a result in submitted data.

this is what expected: 
```javascript
<option value={name: 'John', age: 35}> John </option>
```

**Comment by bluebill1049** [comment, marked as answer]
you will have to build a controlled component and wrapped with Controller, native input only support select as `string`, you can consider to custom component like `react-select`

**Comment by BigSamu** [comment]
Hi,

Do you have an example for this? I am able to do that using a custom Controlled component built with MUI. However I wonder if this is OK or not in term of performance.

Regards,

Samuel
